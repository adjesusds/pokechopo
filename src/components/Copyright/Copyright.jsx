// External Packages
import { node } from 'prop-types'
import React from 'react'

// Internal Packages
import StyledCopyright from './styled/Copyright'

/**
 * Creates a new Copyright component.
 * @class
 */
const Copyright = ({ children }) => <StyledCopyright>{children}</StyledCopyright>

/** @override */
Copyright.propTypes = {
  children: node.isRequired,
}

export default Copyright
