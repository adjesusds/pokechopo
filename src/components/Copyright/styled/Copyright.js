// External Packages
import styled from 'styled-components'

// Internal Packages
import { dimensions } from 'config/palette'

/**
 * @constant {Element} style The style of the component.
 */
const style = styled.span`
  bottom: 0;
  font-size: ${dimensions.xSmall};
  margin: ${dimensions.xxSmall};
  position: absolute;
  right: 0;
`

export default style
