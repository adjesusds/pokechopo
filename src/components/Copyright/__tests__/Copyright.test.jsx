// External Packages
import { shallow } from 'enzyme'
import React from 'react'

// Internal Packages
import Copyright from '../Copyright'

describe('<Copyright />', () => {
  it('should render the component on valid props', () => {
    const component = shallow(<Copyright>Hello, World!</Copyright>)

    expect(component).toMatchSnapshot()
  })
})
