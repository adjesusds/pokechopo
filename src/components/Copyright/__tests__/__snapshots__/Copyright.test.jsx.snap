// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copyright /> should render the component on valid props 1`] = `
<styled.span>
  Hello, World!
</styled.span>
`;
